src/redisearch_rs/rqe_iterators/src/inverted_index.rs

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! Inverted index iterator implementation
+use ffi::t_docId;
+use inverted_index::{IndexReader, NumericReader, RSIndexResult};
+
+use crate::{RQEIterator, RQEIteratorError, SkipToOutcome};
+
+/// A generic full iterator over inverted index entries.
+///
+/// This iterator provides full index scan to all document IDs in an inverted index.
+/// It is not suitable for queries.
+///
+/// # Type Parameters
+///
+/// * `'index` - The lifetime of the index being iterated over.
+/// * `R` - The reader type used to read the inverted index.
+struct FullIterator<'index, R>
+where
+    R: IndexReader<'index>,
```

Comment:
Most Rust code doesn't qualify generics on struct definitions
```suggestion
struct FullIterator<'index, R>
```